packages/@lwc/ssr-compiler/src/__tests__/estemplate.spec.ts

```diff
@@ -33,7 +34,7 @@ describe.each(
         test('when attempting to replace unreplaceable code constructs', () => {
             // Someone might try to create a template where 'class' or 'function'
             // is provided as an argument to the ES template.
-            const isFunctionOrClass = (node: any) =>
+            const isFunctionOrClass = (node: any): node is FunctionDeclaration | ClassDeclaration =>
```

Comment:
This is an error if you're using the workspace version of typescript (5.4.5). Type predicates are only inferred in 5.5 🥲 so close.